[CI] search/180_locale_dependent_mapping/Test Index and Search locale dependent mappings / dates

Failed in a couple periodic builds today, looks like they're all on FIPS but that may just be coincidence

* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java8fips,nodes=immutable&&linux&&docker/294/console
* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java8fips,nodes=immutable&&linux&&docker/294/console
* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java8fips,nodes=immutable&&linux&&docker/69/console
* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java8fips,nodes=immutable&&linux&&docker/69/console

Here's a log from one
[build-periodic-7.x-69.txt.zip](https://github.com/elastic/elasticsearch/files/2959296/build-periodic-7.x-69.txt.zip)

```
REPRODUCE WITH: ./gradlew :distribution:archives:integ-test-zip:integTestRunner -Dtests.seed=DE4EDE4CD5F93BE9 -Dtests.class=org.elasticsearch.test.rest.IntegTestZipClientYamlTestSuiteIT -Dtests.method="test {p0=search/180_locale_dependent_mapping/Test Index and Search locale dependent mappings / dates}" -Dtests.security.manager=true -Dtests.locale=es-PY -Dtests.timezone=Asia/Bahrain -Dcompiler.java=12 -Druntime.java=8FIPS -Djavax.net.ssl.keyStorePassword=password -Djavax.net.ssl.trustStorePassword=password
```

```
  1>             "type" : "parse_exception",
  1>             "reason" : "failed to parse date field [Di, 05 Dez 2000 02:55:00 -0800] with format [E, d MMM yyyy HH:mm:ss Z]: [Text 'Di, 05 Dez 2000 02:55:00 -0800' could not be parsed at index 0]",
  1>             "stack_trace" : "ElasticsearchParseException[failed to parse date field [Di, 05 Dez 2000 02:55:00 -0800] with format [E, d MMM yyyy HH:mm:ss Z]: [Text 'Di, 05 Dez 2000 02:55:00 -0800' could not be parsed at index 0]]; nested: DateTimeParseException[Text 'Di, 05 Dez 2000 02:55:00 -0800' could not be parsed at index 0];
  1>    at org.elasticsearch.common.time.JavaDateMathParser.parseDateTime(JavaDateMathParser.java:233)
  1>    at org.elasticsearch.common.time.JavaDateMathParser.parse(JavaDateMathParser.java:75)
  1>    at org.elasticsearch.index.mapper.DateFieldMapper$DateFieldType.parseToLong(DateFieldMapper.java:397)
  1>    at org.elasticsearch.index.mapper.DateFieldMapper$DateFieldType.isFieldWithinQuery(DateFieldMapper.java:411)
  1>    at org.elasticsearch.index.query.RangeQueryBuilder.getRelation(RangeQueryBuilder.java:447)
  1>    at org.elasticsearch.index.query.RangeQueryBuilder.doRewrite(RangeQueryBuilder.java:464)
  1>    at org.elasticsearch.index.query.AbstractQueryBuilder.rewrite(AbstractQueryBuilder.java:265)
  1>    at org.elasticsearch.search.builder.SearchSourceBuilder.rewrite(SearchSourceBuilder.java:962)
  1>    at org.elasticsearch.search.builder.SearchSourceBuilder.rewrite(SearchSourceBuilder.java:82)
  1>    at org.elasticsearch.index.query.Rewriteable.rewrite(Rewriteable.java:68)
  1>    at org.elasticsearch.index.query.Rewriteable.rewrite(Rewriteable.java:51)
  1>    at org.elasticsearch.search.internal.ShardSearchLocalRequest$RequestRewritable.rewrite(ShardSearchLocalRequest.java:271)
  1>    at org.elasticsearch.search.internal.ShardSearchLocalRequest$RequestRewritable.rewrite(ShardSearchLocalRequest.java:261)
  1>    at org.elasticsearch.index.query.Rewriteable.rewrite(Rewriteable.java:68)
  1>    at org.elasticsearch.search.SearchService.createSearchContext(SearchService.java:659)
  1>    at org.elasticsearch.search.SearchService.createSearchContext(SearchService.java:638)
  1>    at org.elasticsearch.search.SearchService.createContext(SearchService.java:601)
  1>    at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:582)
  1>    at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:385)
  1>    at org.elasticsearch.search.SearchService.access$100(SearchService.java:123)
  1>    at org.elasticsearch.search.SearchService$2.onResponse(SearchService.java:357)
  1>    at org.elasticsearch.search.SearchService$2.onResponse(SearchService.java:353)
  1>    at org.elasticsearch.search.SearchService$4.doRun(SearchService.java:1068)
  1>    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
  1>    at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:43)
  1>    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:747)
  1>    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
  1>    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  1>    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  1>    at java.lang.Thread.run(Thread.java:748)
  1> Caused by: java.time.format.DateTimeParseException: Text 'Di, 05 Dez 2000 02:55:00 -0800' could not be parsed at index 0
  1>    at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1949)
  1>    at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1777)
  1>    at org.elasticsearch.common.time.JavaDateMathParser.parseDateTime(JavaDateMathParser.java:221)
  1>    ... 29 more
```

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/search/180_locale_dependent_mapping.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/search/180_locale_dependent_mapping.yml
@@ -1,8 +1,8 @@
 ---
 "Test Index and Search locale dependent mappings / dates":
   - skip:
-      version: "all"
-      reason: "Awaits fix: https://github.com/elastic/elasticsearch/issues/39981(Previously: JDK9 only supports this with a special sysproperty added in 6.2.0.)"
+      version: " - 6.1.99"
+      reason: JDK9 only supports this with a special sysproperty added in 6.2.0
   - do:
       indices.create:
           index: test_index

